Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp

```diff
@@ -292,6 +292,12 @@ bool WebLoaderStrategy::tryLoadingUsingURLSchemeHandler(ResourceLoader& resource
     if (!webPage || !webFrame)
         return false;
 
+    if (RefPtr documentLoader = resourceLoader.documentLoader()) {
+        auto url = resourceLoader.request().url();
+        if (url.protocolIsAbout() && !documentLoader->isHandledByAboutSchemeHandler())
```

Comment:
Why do we need this?

Reply:
This test `TestWebKitAPI.WebKit.AttrStyle` fails without this check.

Reply:
But the question is why is it needed? Why isn't the check in WebCore sufficient?

Reply:
Oh, I think I get it, the check in WebCore only covers main resource loads. The issue is that `TestWebKitAPI.WebKit.AttrStyle` is trying to do a subresource load to an `about:` URL..

Reply:
This makes me think that there is a bug with the about scheme handler. It should be harmless to go to the about scheme handler, it should respond with an empty response and it should not have a different behavior than if we had no about scheme handler.

Can we investigate why the behavior is different? Maybe calling `didComplete` without calling `didReceiveResponse` is wrong?

Reply:
Got it. I'll dig into this.

Reply:
We need to pass ResourceError do didComplete If we didn't send any response.